### PR TITLE
hip: improve --with-hip-sm option 

### PR DESCRIPTION
**Pull request by yfguo**
## Pull Request Description

Adding option to let HIPCC auto-detect target id. Cleaned up option help strings.

Tested and confirmed fixing building problem on Crusher.

#222 would not be needed once this comes in.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Have read and agree to the Yaksa CLA terms (https://github.com/pmodels/yaksa/wiki/Yaksa-Contributor-License-Agreement)

### CODE REVIEW

**Comment by yfguo** [comment]
Didn't notice there is #222. Will check that out.

File: src/backend/hip/subconfigure.m4 [hunk 1]
```diff
@@ -11,38 +11,23 @@
 # --with-hip-sm
 AC_ARG_WITH([hip-sm],
             [
-  --with-hip-sm=<options> (https://reviews.llvm.org/differential/changeset/?ref=2126544)
+  --with-hip-sm=<options>
           Comma-separated list of below options:
+                auto - letting HIPCC to auto detect target_id
```
Comment (raffenet) [inline src/backend/hip/subconfigure.m4:16]: auto-detection is a great additional. the "all" target took forever to build in my experience.
```suggestion
                auto - let HIPCC auto detect the target_id
```

Comment (hzhou) [reply src/backend/hip/subconfigure.m4:16]: Yanfei and I had discussed offline for a bit. I think we can simply remove the "all" and the subsets options. By default, it is essentially "auto", where `hipcc` do whatever its default. Then user always can pass in specific list of sm, which we directly pass on to `hipcc`. I don't think it is worth while for us to maintain a separate option behavior from `hipcc`'s.

File: src/backend/hip/subconfigure.m4 [hunk 2]
```diff
@@ -121,25 +95,28 @@ if test "${have_hip}" = "yes" ; then
...
+                HIP_GENCODE="--offload-arch=${sm}"
+            else
+                HIP_GENCODE="${HIP_GENCODE} --offload-arch=${sm}"
+            fi
+        done
         if test -z "${HIP_GENCODE}" ; then
-            HIP_GENCODE="--offload-arch=${sm}"
-        else
-            HIP_GENCODE="${HIP_GENCODE} --offload-arch=${sm}"
+            AC_MSG_ERROR([--with-hip-sm not specified; either specify it or disable hip support])
```
Comment (hzhou) [inline src/backend/hip/subconfigure.m4:116]: I am thinking that the default is `auto`.  If the user do `--with-hip-sm=xxx`, we directly pass to `--offload-arch=xxx`.

Comment (yfguo) [reply src/backend/hip/subconfigure.m4]: Yeah, the message here is obsolete. We receive the option as comma-separated list, but HIPCC takes them as separate `--offload-arch=` options.

**Review by hzhou — APPROVED** [review]
LGTM

**Pull request merged**